consensus/metrics.go

```diff
@@ -234,75 +243,96 @@ func PrometheusMetrics(namespace string, labelsAndValues ...string) *Metrics {
 			Help:      "Number of rounds failed on consensus",
 			Buckets:   stdprometheus.LinearBuckets(0, 1, 5),
 		}, labels).With(labelsAndValues...),
-		ProposalCreating: prometheus.NewHistogramFrom(stdprometheus.HistogramOpts{
+		DurationProposal: prometheus.NewHistogramFrom(stdprometheus.HistogramOpts{
+			Namespace: namespace,
+			Subsystem: MetricsSubsystem,
+			Name:      "duration_proposal",
+			Help:      "Duration of proposal step",
+			Buckets:   stdprometheus.LinearBuckets(100, 100, 10),
+		}, labels).With(labelsAndValues...),
+		DurationPrevote: prometheus.NewHistogramFrom(stdprometheus.HistogramOpts{
 			Namespace: namespace,
 			Subsystem: MetricsSubsystem,
-			Name:      "duration_proposal_creating",
-			Help:      "Duration of creating proposal and block",
+			Name:      "duration_prevote",
+			Help:      "Duration of prevote step",
 			Buckets:   stdprometheus.LinearBuckets(100, 100, 10),
 		}, labels).With(labelsAndValues...),
-		ProposalWaiting: prometheus.NewHistogramFrom(stdprometheus.HistogramOpts{
+		DurationPrecommit: prometheus.NewHistogramFrom(stdprometheus.HistogramOpts{
 			Namespace: namespace,
 			Subsystem: MetricsSubsystem,
-			Name:      "duration_proposal_waiting",
-			Help:      "Duration between enterNewRound and receiving proposal",
+			Name:      "duration_precommit",
+			Help:      "Duration of precommit step",
 			Buckets:   stdprometheus.LinearBuckets(100, 100, 10),
 		}, labels).With(labelsAndValues...),
-		ProposalVerifying: prometheus.NewHistogramFrom(stdprometheus.HistogramOpts{
+		DurationCommitExecuting: prometheus.NewHistogramFrom(stdprometheus.HistogramOpts{
 			Namespace: namespace,
 			Subsystem: MetricsSubsystem,
-			Name:      "duration_proposal_verifying",
-			Help:      "Duration of ValidBlock",
-			Buckets:   stdprometheus.LinearBuckets(50, 50, 10),
+			Name:      "duration_commit_executing",
+			Help:      "Duration of executing block txs",
+			Buckets:   stdprometheus.LinearBuckets(100, 100, 10),
 		}, labels).With(labelsAndValues...),
-		ProposalBlockReceiving: prometheus.NewHistogramFrom(stdprometheus.HistogramOpts{
+		DurationCommitCommitting: prometheus.NewHistogramFrom(stdprometheus.HistogramOpts{
 			Namespace: namespace,
 			Subsystem: MetricsSubsystem,
-			Name:      "duration_proposal_block_receiving",
-			Help:      "Duration of receiving all proposal block parts",
+			Name:      "duration_commit_committing",
```

Comment:
About variable names. Looking at the original code, it seems good to compose it with the words that explain Help. Isn't it ``duration_committing_updated_state` instead of `duration_commit_committing`?

Reply:
I'm sorry but this is hard to change. `Grafana` is using this name. This is what happened when I made the name more hierarchical.

Reply:
Where is the name used in Grafana?